diff --git a/include/llvm/ADT/StringRef.h b/include/llvm/ADT/StringRef.h
index 489cb9feb..047c4bf21 100644
--- a/include/llvm/ADT/StringRef.h
+++ b/include/llvm/ADT/StringRef.h
@@ -570,24 +570,4 @@ namespace llvm {
   template <> struct isPodLike<StringRef> { static const bool value = true; };
 }
 
-// HLSL Change Starts
-// StringRef provides an operator string; that trips up the std::pair noexcept specification,
-// which (a) enables the moves constructor (because conversion is allowed), but (b)
-// misclassifies the the construction as nothrow.
-namespace std {
-  template<>
-  struct is_nothrow_constructible <std::string, llvm::StringRef>
-    : std::false_type {
-  };
-  template<>
-  struct is_nothrow_constructible <std::string, llvm::StringRef &>
-    : std::false_type {
-  };
-  template<>
-  struct is_nothrow_constructible <std::string, const llvm::StringRef &>
-    : std::false_type {
-  };
-}
-// HLSL Change Ends
-
 #endif
